src/telliot_feeds/queries/abi_query.py

```diff
@@ -36,11 +37,21 @@ def query_data(self) -> bytes:
 
         This method uses ABI encoding to encode the query's parameter values.
         """
-        param_values = [getattr(self, p["name"]) for p in self.abi]
-        param_types = [p["type"] for p in self.abi]
-        encoded_params = encode_abi(param_types, param_values)
+        # If the query has parameters
+        if self.abi:
+            param_values = [getattr(self, p["name"]) for p in self.abi]
+            param_types = [p["type"] for p in self.abi]
+            encoded_params = encode_abi(param_types, param_values)
 
-        return encode_abi(["string", "bytes"], [type(self).__name__, encoded_params])
+            return encode_abi(["string", "bytes"], [type(self).__name__, encoded_params])
+        # If the query has no parameters
+        else:
+            encoded_params = encode_single("bytes", b"")
+            q_data_hex_str = encode_abi(["string", "bytes"], [type(self).__name__, encoded_params]).hex()
+            q_data_hex_lis = list(q_data_hex_str)
+            q_data_hex_lis[-66] = "2"
+            q_data_hex_str = "".join(q_data_hex_lis)
+            return bytes.fromhex(q_data_hex_str)
```

Comment:
not sure whats going here or whats the query data you're looking for?

Reply:
maybe I can look at the dataspecs

Reply:
I added a comment explainer. Tim is updating the dataspecs about the phantom param..